uiowa/uiowa
# add graphing solution

As a user, I want to be able to use graphs on my Sitenow v3 site.

# How to test

1. Pull this branch and do all the stuff to get it up and running.  
2. Enable the Graph split.

2. create an excel file and add some data to it. It muse be formatted like so, and save it as a .csv file.

| dates or other incremental header  |  dataset1 | dataset2  |
|---|---|---|
| Date1  | foo1  | bar1  |
| Date2  | foo2  | bar2  |
| ... | ...  |  ... |


You can have as many columns or rows as you would like, i just put in this amount for example purposes.

[Here is an example `.csv` for you to use](https://sitenow.uiowa.edu/sites/sitenow.uiowa.edu/files/2021-09/airtravel.csv)

3. open the excel file in a text editor and copy the plain text.
4.  add a "Graph" block to a page in layout builder
5. Give a summary of the data in the summary field.
5. paste the csv text in to the plaintext field
6. save
7. View that your first and second columns of data (that is not dates or whatever you used in column 0) are black and gold, and that any subsequent columns of data are randomly generated colors.
8. Inspect the canvas element that the Graph is drawn in and view that it has `role="img"` and `aria-labeledby="SOME-GRAPH-NAME"`
9. Verify that you can click on `Show tabulated data` and it will show you the table of the data that made the graph, and that the caption is what was entered in the summary field.
10. Verify that the summary field has the  same Id that was listed in the `aria-labeledby` on the canvas.
8. profit.
